src/main/java/edu/ie3/util/geo/GeoUtils.java

```diff
@@ -0,0 +1,342 @@
+   * https://github.com/locationtech/jts/issues/531
+   *
+   * @param lineString the linestring that should be checked and maybe converted to a 'safe to be
+   *     compared' linestring
+   * @return a 'safe to be compared' linestring
+   */
+  public static LineString buildSafeLineString(LineString lineString) {
+    if (lineString.getCoordinates().length == 2) {
+      return buildSafeLineStringBetweenPoints(lineString.getStartPoint(), lineString.getEndPoint());
+    } else {
+      // rebuild line with unique points
+      /* Please note, that using a simple HashSet here to obtain uniqueness is harmful, as it not necessarily maintains
+       * the order of coordinates (but most likely will). Additionally, the behaviour of a HashSet might change with the
+       * JVM (version) you use to execute the code. */
+      Coordinate[] uniqueCoords =
+          Arrays.stream(lineString.getCoordinates()).distinct().toArray(Coordinate[]::new);
+      return uniqueCoords.length == 1
+          ? buildSafeLineStringBetweenPoints(lineString.getStartPoint(), lineString.getEndPoint())
+          : DEFAULT_GEOMETRY_FACTORY.createLineString(uniqueCoords);
+    }
+  }
+
+  /**
+   * Build an instance of {@link LineString} between two points that is safe to be compared even if
+   * the provided two points consist of exactly the same coordinates. This is done by increasing the
+   * coordinate of the provided Point {@code p1} by a small amount to make it different from Point
+   * {@code p2}. For details on the bug inside {@link LineString} that is addressed here, see
+   * https://github.com/locationtech/jts/issues/531
+   *
+   * @param p1 start point of the linestring
+   * @param p2 end point of the linestring
+   * @return a {@link LineString} between the provided points
+   */
+  public static LineString buildSafeLineStringBetweenPoints(final Point p1, final Point p2) {
+    return buildSafeLineStringBetweenCoords(p1.getCoordinate(), p2.getCoordinate());
+  }
+
+  /**
+   * Build an instance of {@link LineString} between two coordinates that is safe to be compared
+   * even if the provided two coordinates are exactly the same coordinates. This is done by
+   * increasing the coordinate of the provided Point {@code c1} by a small amount to make it
+   * different from Point {@code c2}. For details on the bug inside {@link LineString} that is
+   * addressed here, see https://github.com/locationtech/jts/issues/531
+   *
+   * @param c1 start coordinate of the linestring
+   * @param c2 end coordinate of the linestring
+   * @return A safely build line string
+   */
+  public static LineString buildSafeLineStringBetweenCoords(
+      final Coordinate c1, final Coordinate c2) {
+    final Coordinate safeCoord1 = c1.equals(c2) ? buildSafeCoord(c1) : c1;
+    return DEFAULT_GEOMETRY_FACTORY.createLineString(
+        ArrayUtils.addAll(new Coordinate[] {safeCoord1}, c2));
```

Comment:
```suggestion
        ArrayUtils.addAll(safeCoord1, c2));
```

shouldn't this suffice?

Reply:
Found an even better solution imho :)